mxnet-bot/LabelBotFullFunctionality/LabelBot.py

```diff
@@ -0,0 +1,247 @@
+        It checks whether labels exist in the repo, and updates existing labels to the issue
+        :param issue_num: The specific issue number we want to label
+        :param labels: The labels which we want to remove
+        :return Response denoting success or failure for logging purposes
+        """
+        labels = self.format_labels(labels)
+        issue_labels_url = 'https://api.github.com/repos/{repo}/issues/{id}/labels' \
+            .format(repo=self.repo, id=issue_num)
+
+        response = requests.put(issue_labels_url, data=json.dumps(labels), auth=self.auth)
+        if response.status_code == 200:
+            logging.info('Successfully updated labels to {}: {}.'.format(str(issue_num), str(labels)))
+            return "Updated labels successfully"
+        else:
+            logging.error("Could not update the labels")
+            logging.error(response.json())
+            return "Failed to update labels"
+
+    def add_comment(self, issue_num, message):
+        """
+        This method will trigger a comment to an issue by the label bot
+        :param issue_num: The issue we want to comment
+        :param message: The comment message we want to send
+        :return Response denoting success or failure for logging purposes
+        """
+        send_msg = {"body": message}
+        issue_comments_url = 'https://api.github.com/repos/{repo}/issues/{id}/comments' \
+            .format(repo=self.repo, id=issue_num)
+
+        response = requests.post(issue_comments_url, data=json.dumps(send_msg), auth=self.auth)
+        if response.status_code == 201:
+            logging.info('Successfully commented')
+            return "Commented successfully"
+        else:
+            logging.error("Could not comment")
+            return "Failed to comment"
+
+    def parse_webhook_data(self, event):
+        """
+        This method triggers the label bot when the appropriate
+        GitHub event is recognized by use of a webhook
+        :param event: The event data that is received whenever a github issue, issue comment, etc. is made
+        :return: Log statements which we can track in lambda
+        """
+        try:
+            github_event = ast.literal_eval(event["Records"][0]['body'])['headers']["X-GitHub-Event"]
+        except KeyError:
+            raise json.ParseError('Expected a GitHub Event')
+            return "Not a GitHub Event"
+
+        # Grabs actual payload data of the appropriate GitHub event needed for labelling
+        if github_event == "issue_comment":
+            payload = json.loads(ast.literal_eval(event["Records"][0]['body'])['body'])
+
+            # Acquiring labels specific to this repo
+            labels = []
+            if "@mxnet-label-bot" in payload["comment"]["body"]:
+                labels += self.tokenize(payload["comment"]["body"])
+                self.find_all_labels()
+
+                if "@mxnet-label-bot add" in payload["comment"]["body"]:
```

Comment:
can consider tokenizing your payload such that `add`, `remove`, `update` are compared instead of the entire command